src/Microsoft.AspNet.Razor/RazorError.cs

```diff
@@ -46,19 +61,23 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            var err = obj as RazorError;
-            return (err != null) && Equals(err);
+            var error = obj as RazorError;
+            return Equals(error);
         }
 
         public override int GetHashCode()
         {
-            return base.GetHashCode();
+            return HashCodeCombiner.Start()
+                .Add(Message, StringComparer.Ordinal)
+                .Add(Location)
```

Comment:
no, `Length` is not included in `Equals()` calculation. so equal values with different `Length`s would have different hash codes. we have an answer for that case: don't do that.